File: CampingNeretva/UI/campingneretva_desktop/lib/models/reservation_model.dart
import 'user_model.dart';
import 'parcel_model.dart';
import 'activity_model.dart';
import 'reservation_accommodation_model.dart';
import 'reservation_person_model.dart';
import 'reservation_rentable_item_model.dart';
import 'reservation_vehicle_model.dart';

class Reservation {
  final int reservationId;
  final DateTime startDate;
  final DateTime endDate;
  final User user;
  final Parcel parcel;
  final List<ReservationPersonModel> persons;
  final List<ReservationVehicleModel> vehicles;
  final List<ReservationAccommodationModel> accommodations;
  List<ReservationRentableItemModel>? rentableItems;
  List<Activity>? activities;
  final double totalPrice;
  final String paymentStatus;

  Reservation({
    required this.reservationId,
    required this.startDate,
    required this.endDate,
    required this.user,
    required this.parcel,
    required this.persons,
    required this.vehicles,
    required this.accommodations,
    this.rentableItems,
    this.activities,
    required this.totalPrice,
    required this.paymentStatus,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) {
    return Reservation(
      reservationId: json['reservationId'],
      startDate: DateTime.parse(json['checkInDate']),
      endDate: DateTime.parse(json['checkOutDate']),
      user: User.fromJson(json['user']),
      parcel: Parcel.fromJson(json['parcel']),
      persons:
          (json['reservationPeople'] as List)
              .map((e) => ReservationPersonModel.fromJson(e))
              .toList(),
      vehicles:
          (json['reservationVehicles'] as List)
              .map((e) => ReservationVehicleModel.fromJson(e))
              .toList(),
      accommodations:
          (json['reservationAccommodations'] as List)
              .map((e) => ReservationAccommodationModel.fromJson(e))
              .toList(),
      rentableItems:
          json['reservationRentables'] == null
              ? null
              : (json['reservationRentables'] as List)
                  .map(
                    (e) => ReservationRentableItemModel.fromJson(
                      e as Map<String, dynamic>,
                    ),
                  )
                  .toList(),
      activities:
          json['activities'] == null
              ? null
              : (json['activities'] as List)
                  .map((e) => Activity.fromJson(e as Map<String, dynamic>))
                  .toList(),
      totalPrice: json['totalPrice'].toDouble(),
      paymentStatus: json['paymentStatus'],
    );
  }
}

File: CampingNeretva/CampingNeretva.Model/Models/ReservationModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CampingNeretva.Model.Models
{
    public class ReservationModel
    {
        public int ReservationId { get; set; }
        public int UserId { get; set; }
        public int ParcelId { get; set; }
        public DateTime CheckInDate { get; set; }
        public DateTime CheckOutDate { get; set; }
        public decimal TotalPrice { get; set; }
        public string PaymentStatus { get; set; }
        public ParcelModel Parcel { get; set; }
        public UserModel User { get; set; }
        public List<ReservationPersonModel> ReservationPeople { get; set; }
        public List<ReservationVehicleModel> ReservationVehicles { get; set; }
        public List<ReservationAccommodationModel> ReservationAccommodations { get; set; }
        public List<ReservationRentableItemModel> ReservationRentables { get; set; }
        public List<ActivityModel> Activities { get; set; }
    }
}
